### PR TITLE
Remove .NET Framework 4.6.1 support 

### LINKED ISSUE #376: Remove .NET Framework 4.6.1 support
https://devblogs.microsoft.com/dotnet/net-framework-4-5-2-4-6-4-6-1-will-reach-end-of-support-on-april-26-2022/

_Originally posted by @Kielek in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/375#issuecomment-1046641141_

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #376

Changes proposed in this pull request:
Drop support for .NET Framework 4.6.1 - it is EOS within 2 months.
Bump LangVersion to 10.0 also for native tests.
Change windows-2019 to windows-2022 in our CI pipeline.

### CODE REVIEW

**Comment by Kielek** [comment]
Guys, it will be great if you can switch required build from windows-2019 to windows-2022. Without this change this PR cannot be merged.

**Comment by pellared** [comment]
Removed `build (windows-2019)` as a required job to make it possible to merge it.

**Comment by Kielek** [comment]
I have executed integration tests on branch for .NET 4.6.2. All passed. 
```
  14:19:31 [INF] > C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\dotnet.exe test D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.MongoDB\IntegrationTests.MongoDB.csproj --configuration Release --framework net462 --logger trx --no-build --no-restore --results-directory D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.MongoDB /property:Platform=x64
  14:19:31 [INF] > C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\dotnet.exe test D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.StartupHook\IntegrationTests.StartupHook.csproj --configuration Release --framework net462 --logger trx --no-build --no-restore --results-directory D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.StartupHook /property:Platform=x64
  14:19:31 [INF] > C:\Users\runneradmin\AppData\Local\Microsoft\dotnet\dotnet.exe test D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.GraphQL\IntegrationTests.GraphQL.csproj --configuration Release --framework net462 --logger trx --no-build --no-restore --results-directory D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.GraphQL /property:Platform=x64
  14:20:44 [DBG] Test run for D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.MongoDB\bin\x64\Release\net462\IntegrationTests.MongoDB.dll (.NETFramework,Version=v4.6.2)
  14:20:44 [DBG] Microsoft (R) Test Execution Command Line Tool Version 17.1.0
  14:20:44 [DBG] Copyright (c) Microsoft Corporation.  All rights reserved.
  14:20:44 [DBG] 
  14:20:44 [DBG] Starting test execution, please wait...
  14:20:44 [DBG] A total of 1 test files matched the specified pattern.
  14:20:44 [DBG] Results File: D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.MongoDB\runneradmin_fv-az90-160_2022-02-24_14_20_44.trx
  14:20:44 [DBG] 
  14:20:44 [DBG] Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration:  - IntegrationTests.MongoDB.dll (net462)
  14:20:44 [DBG] Test run for D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\test\integration-tests\IntegrationTests.GraphQL\bin\x64\Release\net462\IntegrationTests.GraphQL.dll (.NETFramework,Version=v4.6.2)
  14:20:44 [DBG] Microsoft (R) Test Execution Command Line Tool Version 17.1.0
  14:20:44 [DBG] Copyright (c) Microsoft Corporation.  All rights reserved.
  14:20:44 [DBG] 
  14:20:44 [DBG] Starting test execution, please wait...
  14:20:44 [DBG] A total of 1 test files matched the specified pattern.
  14:20:44 [DBG] Results File: D:\a\opentelemetry-dotnet-instrumentation\opentelemetry-dotnet-instrumentation\build_data\results\IntegrationTests.GraphQL\runneradmin_fv-az90-160_2022-02-24_14_19_51.trx
  14:20:44 [DBG] 
  14:20:44 [DBG] Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration:  - IntegrationTests.GraphQL.dll (net462)
```

Unfortunately, there is some issue to execute tests for all frameworks together. It is tracked under #378

**Review by pellared — APPROVED** [review]

File: samples/ConsoleApp.SelfBootstrap/ConsoleApp.SelfBootstrap.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>net461;netcoreapp3.1;net5.0</TargetFrameworks>
+    <TargetFrameworks>net462;netcoreapp3.1;net5.0</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (Kielek) [reply samples/ConsoleApp.SelfBootstrap/ConsoleApp.SelfBootstrap.csproj:5]: Lets keep this PR small.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by nrcventura — APPROVED** [review]

**Pull request merged**